packages/flutter/lib/src/material/bottom_navigation_bar_theme.dart

```diff
@@ -0,0 +1,261 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show lerpDouble;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'bottom_navigation_bar.dart';
+import 'theme.dart';
+
+/// Defines default property values for descendant [BottomNavigationBar]
+/// widgets.
+///
+/// Descendant widgets obtain the current [BottomNavigationBarThemeData] object
+/// using `BottomNavigationBarTheme.of(context)`. Instances of
+/// [BottomNavigationBarThemeData] can be customized with
+/// [BottomNavigationBarThemeData.copyWith].
+///
+/// Typically a [BottomNavigationBarThemeData] is specified as part of the
+/// overall [Theme] with [ThemeData.bottomNavigationBarTheme].
+///
+/// All [BottomNavigationBarThemeData] properties are `null` by default. When
+/// null, the [BottomNavigationBar] constructor provides defaults.
+///
+/// See also:
+///
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+@immutable
+class BottomNavigationBarThemeData with Diagnosticable {
+  /// Creates a theme that can be used for [ThemeData.BottomNavigationBarTheme].
+  const BottomNavigationBarThemeData({
+    this.backgroundColor,
+    this.elevation,
+    this.selectedIconTheme,
+    this.unselectedIconTheme,
+    this.selectedItemColor,
+    this.unselectedItemColor,
+    this.selectedLabelStyle,
+    this.unselectedLabelStyle,
+    this.showSelectedLabels,
+    this.showUnselectedLabels,
+    this.type,
+  });
+
+  /// The color of the [BottomNavigationBar] itself.
+  ///
+  /// See [BottomNavigationBar.backgroundColor].
+  final Color backgroundColor;
+
+  /// The z-coordinate of the [BottomNavigationBar].
+  ///
+  /// See [BottomNavigationBar.elevation].
+  final double elevation;
+
+  /// The size, opacity, and color of the icon in the currently selected
+  /// [BottomNavigationBarItem.icon].
+  ///
+  /// See [BottomNavigationBar.selectedIconTheme].
+  final IconThemeData selectedIconTheme;
```

Comment:
We talked about this offline, it does make sense for the `BottomNavigationBar` to resolve these values if they are `MaterialStateProperty`s, similar to how raw material button does. Doing this can be done retroactively though, and I'd prefer not to do it as part of this change.